### ISSUE TITLE
Should new_targets use zeros or arange?

### ISSUE THREAD

**Issue by ajaech** [leimao/Sampled_Softmax_PyTorch#1]
In the sampled softmax loss when you create new_targets, should it be using torch.zeros or torch.arange? I thought it should be arange because the label for the ith item in the batch should be in the ith position, right?

`new_targets = Variable(torch.zeros(batch_size).long())`

https://github.com/leimao/Sampled_Softmax_PyTorch/blob/9545b61f425e3ad3790deb6a3d7b9d9d0a601fa5/utils.py#L225

**Comment by cheershuaizhao** [comment]
No, it seems not in the ith position. The true labels are all in the first position.
https://github.com/leimao/Sampled-Softmax-PyTorch/blob/master/utils.py#L224
